### ISSUE TITLE
Using a custom topology in a species tree for simulations

### ISSUE THREAD

**Issue by sivico26** [david-schaller/AsymmeTree#5]
Hi again,

Is it possible to load a custom topology as a species tree (instead of simulating it) and continue with the simulation of gene trees, sequences, and genomes? _ found that I can load Newick with Tralda's `Tree` data structure, but when I try to use it as a species tree, it complains (reasonably so) that the nodes do not have the event attribute defined. In all likelihood, other attributes must be initialized to set it up correctly. 

In short, how can I set up a `Tree` loaded from a Newick to use it as a species tree for simulations?

Thank you in advance for your help.

**Comment by david-schaller** [comment]
Hi,

This should be possible with a few tricks. After loading the Newick tree, you can use the `preorder()` function of the `Tree` instance to iterate over the nodes and set the `event` attribute to `"S"` everywhere (this is fine for a species tree, I assume your species tree does not contain losses?). Also you should give all inner nodes a unique `label` if your Newick tree does not have labels for inner nodes.

Additionally, you need to set the `tstamp` (and maybe `dist` to be on the safe side). This might be a bit tricky. Does the tree you want to load come with distances? Then you could use these to calculate a corresponding timing.
If not, you need to simulate a timing. AsymmeTree has a function `random_ultrametric_timing()` in `asymmetree.tools.PhyloTreeTools` which produces some timing that would work for simulating gene trees. The method is from a very old version of AsymmeTree though and the sampling of time stamps is not what is currently implemented in the species trees simulation (there we use now birth-death processes, i.e. sampling from exponential distributions). Alternative, you could adapt the code in function `_BDP_n_age` in `asymmetree.treeevolve.SpeciesTree` (l. 724 ff) to simulate a timing.

In general, I'm happy to help with this and maybe refactor the code a bit, but I have only very little time in the next weeks.

**Comment by sivico26** [comment]
Hi @david-schaller,

Thanks for the comment; it helped. I managed to load my newick and set the attributes to make it work. In case anyone is interested, I will briefly describe it for future reference.

My Newick tree did have distances, loaded in the nodes' `dist` attribute, and had no dead ends, as you guessed. But I had to modify it to make it like the usual species trees simulations (make it a _planted tree_ on the root; this might not be necessary, but I preferred to stick to it). The tree was not ultrametric, so I had to make it so from the distance attributes. I made a simple iterative function to reconcile the timestamp of an ancestor from the time stamps and distances of its children. Finally, I also added the event attribute ("S") to all the nodes (except the root) and gave names (numbers) to the inner nodes (which were missing). That was it. 

I hope it helps.
